File: Ruby/reverse.rb
def reverse(num,rev)
	if num > 0 
	    rem = (num % 10);
		rev = rev*10+rem;
		reverse(num / 10,rev);
	else
	    return rev;    
	end
end

print "Enter number: ";
number = gets.chomp.to_i;  

result = reverse(number, 0);
print "Result is: ",result;

File: Java/reverse.java
public class Main
{
public static void Reverse(int num)
    {
      
        if (num < 10) {
            System.out.println(num);
            return;
        }
        else {
            System.out.print(num % 10);
            Reverse(num / 10);
        }
    }

    public static void main(String args[])
    {
        int num = 1099;
        System.out.print("Reversed Number: ");
        Reverse(num);
    }
}
